packages/vite/src/node/server/middlewares/hostCheck.ts

```diff
@@ -37,8 +37,12 @@ export function getAdditionalAllowedHosts(
   // allow server origin by default as that indicates that the user is
   // expecting Vite to respond on that host
   if (resolvedServerOptions.origin) {
-    const serverOriginUrl = new URL(resolvedServerOptions.origin)
-    list.push(serverOriginUrl.hostname)
+    // some frameworks may pass the origin as a placeholder, so it's not
+    // possible to parse as URL, so use a try-catch here as a best effort
+    try {
+      const serverOriginUrl = new URL(resolvedServerOptions.origin)
+      list.push(serverOriginUrl.hostname)
+    } catch {}
```

Comment:
should this have a debug log at least? quiet swallowing can lead to long and frustrating debug sessions

Reply:
we can add one, yes, but I don't expect that this fix will stay in place